src/window.ts

```diff
@@ -64,13 +76,41 @@ export function getCachedReferences(window: Window & typeof globalThis): CachedR
     return record;
 }
 
+export function getCachedRedReferences(window: Window & typeof globalThis): CachedRedReferencesRecord {
+    let record: CachedRedReferencesRecord | undefined = WeakMapGet(cachedRedGlobalMap, window);
```

Comment:
extract the base functionaility into a method that can be called for the regular stuff, then add the specific blue and red using typescript type inheritance.

Reply:
Updated. I dried it out.